### ISSUE TITLE
Model @property and SlugRelatedField not handled properly

### ISSUE THREAD

**Issue by sparktx-adam-gleason** [tfranzel/drf-spectacular#943]
In `v0.24.2` this works properly but `v0.25.0` and onward broke how SlugRelatedField and `@properties` are being handled.

**Describe the bug**
`SlugRelatedField` that reference a model `@property` are not being handled.

While stepping through the functions on how this field is being resolved, I tracked down how this field is being handled in `v0.25.0`.

First, this logic was updated in `v0.25.0` compared to `v0.24.2`.
https://github.com/tfranzel/drf-spectacular/blob/master/drf_spectacular/openapi.py#L663

The SlugRelatedField will get to this function. https://github.com/tfranzel/drf-spectacular/blob/7392ba99fc9017c21d571e354c87682b870d26c7/drf_spectacular/plumbing.py#L500

and will finally return this value at the end of traversal.
https://github.com/tfranzel/drf-spectacular/blob/7392ba99fc9017c21d571e354c87682b870d26c7/drf_spectacular/plumbing.py#L508-L509

Ultimately, this assertion is failing. https://github.com/tfranzel/drf-spectacular/blob/0.25.0/drf_spectacular/openapi.py#L544

Prior to this logic update, `SlugRelatedFields` used this logic to resolve. https://github.com/tfranzel/drf-spectacular/blob/0.24.2/drf_spectacular/openapi.py#L698-L699

**To Reproduce**

```python
class MyModel(models.Model):
    label = models.CharField(
        max_length=100,
        unique=True,
    )

    class Meta:
        verbose_name = _("MyModel")
        verbose_name_plural = _("MyModels")

    def __str__(self):
        return self.label

    @property
    def property_field(self):
        return "42"

# How the field is defined in a Serializer
forty_two = serializers.SlugRelatedField(
    slug_field="property_field",
    read_only=True,
)
```

API Response
```json
{
    "forty_two": "42"
}
```

**Expected behavior**
SlugRelatedFields that use @properties don't fail assertion and are resolved the same way they are handled in `v0.24.2`

**Comment by tfranzel** [comment]
I see, will look into it. The change happened here #854. 

Before, we had it hard-coded as string, so there was little chance of it breaking. Now, the introspection is assuming a model, but it looks like a property is also viable. We have the mechanics for that, but somehow it is not triggered.

I'm not exactly sure why you chose a slug here, but we should support it if it works. In the meantime, this should work too (if you add a return type `property_field(self):-> str`): 

```python
forty_two = serializers.ReadOnlyField(source='property_field')
```

**Comment by sparktx-adam-gleason** [comment]
@tfranzel For my use case, I am using `SlugRelatedField` because the actual property value is sourced from a `ForeignKey` related field to my model which I'm building a ModelSerializer for.

`my_model.related_model.property_field`

If I point the `SlugRelatedField` to a non-property field on the related model, I don't see an error in schema generation. This is to be expected because the assertion passes.

**Comment by tfranzel** [comment]
> For my use case, I am using SlugRelatedField because the actual property value is sourced from a ForeignKey related 

Yes, I get that. `ReadOnlyField` also supports relations. What I meant was `serializers.ReadOnlyField(source='related_model.property_field')`

> If I point the SlugRelatedField to a non-property field on the related model, I don't see an error in schema generation. 

Sure, because the mechanics falsely expect a model field for `SlugRelatedField` target. 

Anyways, this fix should do the trick. Please test it if you don't mind.

**Comment by tfranzel** [comment]
closed with release `0.26.0`
